packages/x-components/src/x-modules/history-queries/components/__tests__/my-history.spec.ts

```diff
@@ -61,13 +68,60 @@ describe('testing MyHistory component', () => {
     expect(wrapper.html()).toEqual('');
   });
 
-  it('renders the list of searched queries', async () => {
-    const { search, getListItems } = renderMyHistory();
+  it('renders the list of searched queries group by date', () => {
+    const historyQueriesGroupByDate: HistoryQuery[] = [
+      {
+        query: 'lego',
+        timestamp: 1650286901802,
+        modelName: 'HistoryQuery'
+      },
+      {
+        query: 'barbie',
+        timestamp: 1650286895254,
+        modelName: 'HistoryQuery'
+      },
+      {
+        query: 'truck',
+        timestamp: 1649230515242,
+        modelName: 'HistoryQuery'
+      },
+      {
+        query: 'doll',
+        timestamp: 1649230513535,
+        modelName: 'HistoryQuery'
+      }
+    ];
+
+    const expectedResult: Record<string, HistoryQuery[]> = {
```

Comment:
I would suggest a little refactor to this test.

- Use the data to iterate. right now if the component fails to render no queries, this test still passes.
- Remove the `my-history-query` and `my-history-time` data-test and use a single assertion to test everything (just to simplify it a little bit).
- Inline some variables in function calls to reduce verbosity.

```ts
    const { findAllInWrapper } = renderMyHistory({
      historyQueries: historyQueriesGroupByDate
    });

    const historyWrappers = findAllInWrapper('my-history-item');

    forEach(
      {
        'Monday, April 18, 2022': [historyQueriesGroupByDate[0], historyQueriesGroupByDate[1]],
        'Wednesday, April 6, 2022': [historyQueriesGroupByDate[2], historyQueriesGroupByDate[3]]
      },
      (date, historyQueries, index) => {
        const groupWrapper = historyWrappers.at(index);
        const historyItemWrappers = groupWrapper.findAll(getDataTestSelector('history-query-item'));
        expect(groupWrapper.find(getDataTestSelector('my-history-date')).text()).toBe(date);
        historyQueries.forEach((historyQuery, historyQueryIndex) => {
          const hour = new Date(historyQuery.timestamp).toLocaleTimeString([], {
            hour: '2-digit',
            minute: '2-digit'
          });
          expect(historyItemWrappers.at(historyQueryIndex).text()).toEqual(
            `${historyQuery.query} ${hour} ✕`
          );
        });
      }
    );
```